### DISCUSSION TITLE
How to use GrpcClient with multiple channel

### DISCUSSION THREAD

**Discussion by kmlsharma53** [LogNet/grpc-spring-boot-starter#172] [Q&A]
Hi,

As i understand from gRPC doc : it creates one TCP connection underline (per channel).

Lets say we have 15k request per second on a channel, Do we need to create Pool of ManagedChannel at client side to achieve the high throughput  or single ManagedChannel will be enough (or it does pooling internally)?

Thanks for the help.

**Comment by jvmlet** [comment]
You probably asking in the wrong place. This library does not provide starter for the client (yet)  😉

**Reply by kmlsharma53** [reply]
Sorry, updating my question

**Reply by swarupdonepudi** [reply]
so you have a springboot app that uses this grpc-starter and makes rpc requests to other grpc server? and you are expecting that this client of your sends around 15k rps?

your app(client) sends 15k rps -> grpc server 

did i understand your question correctly?

**Comment by kmlsharma53** [comment]
Yeah, you're right.

Thanks,
Kamal

**Reply by swarupdonepudi** [reply]
ok, so basically you are talking about client side loadbalancing for http2 streams. this starter currently does not really help you with the client side loadbalancing at this point. are you running both your client and server on kubernetes or outside kubernetes?

the client side loadbalancing is pretty tricky concept in grpc. so, the solutions depend heavily on your environment.
